### ISSUE TITLE
Last letter eaten away on some words

### ISSUE THREAD

**Issue by Chaman-veteran** [Chaman-veteran/spell-checker#3]
# Summary
Some words are not printed correctly and the last letter don't appear.

# Expected behavior
All etters should be printed wholly.

# How to reproduce
To easily reproduce it : 
1. Reverse the order in `completeWord` (`Main.hs`) with :
```hs
completeWord tree prefixe = take 10 . sort $ giveSuffixe tree prefixe
```
2. Start the spell checker and type `t` then use tab to complete the word
3. spell-checker prints `tRN` while it's not in the dictionary nor in Sherlock Holmes but `tRNA` is

**Comment by Chaman-veteran** [comment]
When a word is identified, we can check it with usual ordering, it gives, when correcting :
```sh
> tRN
[SRN,BRM,CNN,CRC,CRM,CRT,DRM,EAN,ERM,HRH]
> tRNA
[tRNA,rRNA,mRNA,rDNA,ARPA,WRNS,cDNA]
```
And when completing a word :
```sh
> tRN
[tRNA,tRN]
```
As it's `tRN` that shouldn't be there, it *seems* that the bug appears only when completing a word

**Comment by Chaman-veteran** [comment]
Fixed, the output is still
```sh
> tRN
[tRNA,tRN]
```
on complete, but doesn't eat letters when reverse order is used.
The issue was that when getting suffixes for words, we didn't check if they existed. All possible words to complete what was written was composed of all prefixe++suffixes of existing words containing prefixe++suffixes that do not exist.
We couldn't see it with used words, as we ordered them by frequency.

**Comment by Chaman-veteran** [comment]
Fixed, the output is still
```sh
> tRN
[tRNA,tRN]
```
on complete, but doesn't eat letters when reverse order is used.
The issue was that when getting suffixes for words, we didn't check if they existed. All possible words to complete what was written was composed of all prefixe++suffixes of existing words containing prefixe++suffixes that do not exist.
We couldn't see it with used words, as we ordered them by frequency.
